### PR TITLE
Add gen2 VHD builder pipeline

### PR DESCRIPTION
**Pull request by xuto2**

### CODE REVIEW

File: .pipelines/.vsts-vhd-builder.yaml [hunk 1]
```diff
@@ -80,7 +81,26 @@ phases:
           -e OFFER_NAME="Ubuntu" \
           ${CONTAINER_IMAGE} make  -f packer.mk generate-sas
         displayName: Getting Shared Access Signature URI
-        condition: eq(variables.DRY_RUN, 'False')
+        condition: and(eq(variables.DRY_RUN, 'False'), eq(variables.HYPERV_GENERATION, 'V1'))
+      - script: |
+          subscription_id="$(cat vhdbuilder/packer/settings.json | grep "subscription_id" | awk -F':' '{print $2}' | awk -F'"' '{print $2}')" && \
```
Comment (adilliadil) [inline .pipelines/.vsts-vhd-builder.yaml:86]: Why are we generating these here and passing them to the pacler, can't we just pass them in as input variables

File: vhdbuilder/packer/create-managed-disk-from-sig-gen2.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+for v in "${required_env_vars[@]}"
+do
+    if [ -z "${!v}" ]; then
+        echo "$v was not set!"
+        exit 1
+    fi
+done
+
+sig_resource_id="/subscriptions/${subscription_id}/resourceGroups/${resource_group_name}/providers/Microsoft.Compute/galleries/PackerSigGallery/images/1804Gen2/versions/1.0.${create_time}"
+disk_resource_id="/subscriptions/${subscription_id}/resourceGroups/${resource_group_name}/providers/Microsoft.Compute/disks/1.0.${create_time}"
```
Comment (adilliadil) [inline vhdbuilder/packer/create-managed-disk-from-sig-gen2.sh:21]: Does Packer for Gen2 support how many replicacount, specifying regions for sig publish

File: vhdbuilder/packer/create-managed-disk-from-sig-gen2.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+for v in "${required_env_vars[@]}"
+do
+    if [ -z "${!v}" ]; then
+        echo "$v was not set!"
+        exit 1
+    fi
+done
+
+sig_resource_id="/subscriptions/${subscription_id}/resourceGroups/${resource_group_name}/providers/Microsoft.Compute/galleries/PackerSigGallery/images/1804Gen2/versions/1.0.${create_time}"
```
[the remark this answers is not in the document]
Comment (xuto2) [reply vhdbuilder/packer/create-managed-disk-from-sig-gen2.sh:20]: good question. This is not the final image.
To get final PIR, we need to export the managed disk created here and download it to local, rename it and then upload it to our classic storage account. And then generate a sas url for publishing to PIR. Not automating these manually step as lack of time and also it's expected that Packer/PIR will do something to fill the gap. If this turns out not practical, we can fill the gap of automation 

To get final SIG, will need additional work and my goal is to let it go through the SIG release pipeline.

Make sense?

File: vhdbuilder/packer/create-managed-disk-from-sig-gen2.sh (hunk 2 above)
Comment (xuto2) [reply vhdbuilder/packer/create-managed-disk-from-sig-gen2.sh:21]: again, this is not the final SIG. I intentionally want to separate the build and release of SIG in two pipelines so we have less garbage in our prod SIG. Will see how it goes though.

File: .pipelines/.vsts-vhd-builder.yaml (hunk 1 above)
Comment (xuto2) [reply .pipelines/.vsts-vhd-builder.yaml:86]: we can but we're currently not passing sub_id as input parameter. Instead it's retrieved in the init-variables.sh

SUBSCRIPTION_ID="${SUBSCRIPTION_ID:-$(az account show -o json --query="id" | tr -d '"')}"

If we want we can refactor both, but I don't feel it does not worthy the effort.

File: vhdbuilder/packer/create-managed-disk-from-sig-gen2.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+disk_resource_id="/subscriptions/${subscription_id}/resourceGroups/${resource_group_name}/providers/Microsoft.Compute/disks/1.0.${create_time}"
+
+curl -sL https://github.com/yangl900/armclient-go/releases/download/v0.2.3/armclient-go_linux_64-bit.tar.gz | tar xz
+
+~/armclient put ${disk_resource_id}?api-version=2019-11-01 "{'location': '$location', \
```
Comment (xuto2) [inline vhdbuilder/packer/create-managed-disk-from-sig-gen2.sh:25]: FYI @bowang-666

File: .pipelines/.vsts-vhd-builder.yaml (hunk 1 above)
Comment (xuto2) [reply .pipelines/.vsts-vhd-builder.yaml:86]: Give it a second thought - you're right, the value is already there. Will fix it in the next PR.

**Pull request merged**